test-suite/src/test/scala/org/scalajs/testsuite/compiler/ReflectionTest.scala

```diff
@@ -76,4 +114,10 @@ object ReflectionTest extends JasmineTest {
 
   class RenamedTestClass
 
+  @JSName("ReflectionTestRawJSClass")
```

Comment:
It *is* different, the original name is `ReflectionTest$RefectionTestRawJSClass`.